src/main/java/graphql/execution/Async.java

```diff
@@ -26,15 +26,17 @@
     public static <U> CompletableFuture<List<U>> each(List<CompletableFuture<U>> futures) {
         CompletableFuture<List<U>> overallResult = new CompletableFuture<>();
 
+        @SuppressWarnings("unchecked")
+        CompletableFuture<U>[] arrayOfFutures = futures.toArray(new CompletableFuture[0]);
```

Comment:
Change `futures.toArray(new CompletableFuture[0])`
to `futures.toArray(new CompletableFuture[futures.size()])` will be more efficient, since this will avoid create a new array object.

`toArray()` in `ArrayList`:
```java
    public <T> T[] toArray(T[] a) {
        if (a.length < size)
            // Make a new array of a's runtime type, but my contents:
            return (T[]) Arrays.copyOf(elementData, size, a.getClass());
        System.arraycopy(elementData, 0, a, 0, size);
        if (a.length > size)
            a[size] = null;
        return a;
    }
```

Reply:
Actually see : https://shipilev.net/blog/2016/arrays-wisdom-ancients/